stan/math/prim/prob/wiener4_lcdf.hpp

```diff
@@ -0,0 +1,743 @@
+    const auto two_avw = 2 * a * v * w;
+    const auto two_av = 2 * a * v;
+    const auto exp_two_va_one_minus_w = exp(two_va_one_minus_w);
+    const auto exp_two_avw = exp(two_avw);
+    const auto exp_two_av = exp(two_av);
+    if (((exp_two_va_one_minus_w >= nearly_one) || (exp_two_avw >= nearly_one))
+        || (exp_two_av >= nearly_one)) {
+      return ret_t(-w);
+    }
+    prob = LOG_TWO + two_va_one_minus_w - log1p(-exp_two_va_one_minus_w);
+    auto log_quotient = log1p(-exp_two_avw) - log1p(-exp_two_av);
+    if (log(w) > log_quotient) {
+      prob += log_diff_exp(log(w), log_quotient);
+      return exp(prob);
+    } else {
+      prob += log_diff_exp(log_quotient, log(w));
+      return -exp(prob);
+    }
+  } else {
+    const auto minus_two_va_one_minus_w = (-2.0 * v * a * (1.0 - w));
+    const auto exp_minus_two_va_one_minus_w = exp(minus_two_va_one_minus_w);
+    const auto minus_two_av = (-2 * a * v);
+    const auto exp_minus_two_av = exp(minus_two_av);
+    if ((exp_minus_two_va_one_minus_w >= nearly_one)
+        || (exp_minus_two_av >= nearly_one)) {
+      return ret_t(-w);
+    }
+    prob = LOG_TWO - log1p(-exp_minus_two_va_one_minus_w);
+    ret_t log_quotient;
+    if (minus_two_va_one_minus_w > minus_two_av) {
+      log_quotient = log_diff_exp(minus_two_va_one_minus_w, minus_two_av)
+                     - log1p(-exp_minus_two_av);
+    } else if (minus_two_va_one_minus_w < minus_two_av) {
+      log_quotient = log_diff_exp(minus_two_av, minus_two_va_one_minus_w)
+                     - log1p(-exp_minus_two_av);
+    } else {
+      log_quotient = NEGATIVE_INFTY - log1p(-exp_minus_two_av);
+    }
+    if (log(w) > log_quotient) {
+      prob += log_diff_exp(log(w), log_quotient);
+      return -exp(prob);
+    } else {
+      prob += log_diff_exp(log_quotient, log(w));
+      return exp(prob);
+    }
+  }
+}
+
+/**
+ * Helper function. Log of Mill's ratio for the normal distribution
+ *
+ * @param x A scalar
+ * @return Log of Mill's ratio
+ */
+template <typename T_x>
+inline auto logMill(T_x&& x) {
+  return std_normal_lcdf(-x) - std_normal_lpdf(x);
+}
+
+/**
+ * Calculate the wiener4 distribution
+ *
+ * @tparam NaturalScale Whether to return the distribution on natural or
+ * log-scale
+ *
+ * @param y A scalar variable; the reaction time in seconds
+ * @param a The boundary separation
+ * @param vn The relative starting point
+ * @param wn The drift rate
+ * @param wildcard This parameter space is needed for a functor. Could be
+ * deleted when another solution is found
+ * @param err The log error tolerance
+ * @return distribution
+ */
+template <bool NaturalScale = false, typename T_y, typename T_a, typename T_w,
+          typename T_v, typename T_wildcard, typename T_err>
+inline auto wiener4_distribution(const T_y& y, const T_a& a, const T_v& vn,
+                                 const T_w& wn, T_wildcard&& wildcard = 0.0,
+                                 T_err&& err = log(1e-12)) {
```

Comment:
We need to remove wildcard here and in the other functions

It looks like it is always given a value of 0 everywhere? For example in the below. Does this function need to be estimated with error correction? It looks like the error term there is the 6th argument, but it is never adjusted in the error correction function.

```c++
    const T_partials_return log_cdf
        = internal::estimate_with_err_check<5, 0, GradientCalc::OFF,
                                            GradientCalc::OFF>(
            [](auto&&... args) {
              return internal::wiener4_distribution<GradientCalc::OFF>(args...);
            },
            log_error_cdf - LOG_TWO, y_value - t0_value, a_value, v_value,
            w_value, 0.0, log_error_absolute);
```


If you don't need error correction here then you can just change all of these to remove the first argument and the 0.0.

```cpp
internal::wiener4_distribution<GradientCalc::OFF>(y_value - t0_value, a_value, v_value,
            w_value, log_error_absolute)
```

If you need error correction you can still delete wildcard and use the first int template argument to specify which argument in the function holds the error correction portion.

```c++
    const T_partials_return log_cdf
        = internal::estimate_with_err_check<4, 0, GradientCalc::OFF,
                                            GradientCalc::OFF>(
            [](auto&&... args) {
              return internal::wiener4_distribution<GradientCalc::OFF>(args...);
            },
            log_error_cdf - LOG_TWO, y_value - t0_value, a_value, v_value,
            w_value, log_error_absolute);
```

Reply:
I first inserted the wildcard parameter as I used a function from the PDF code. And in that code one more argument was needed. Therefore I inserted the wildcard as a placeholder. It is not used anywhere. If you have another idea how to solve this problem we can change the code.

But I think the last changes brought some errors. Tests do not pass at the moment.

Reply:
I have a solution! It works without the wildcards. I will push it soon. First, I check the error messages.